spiceypy/spiceypy.py

```diff
@@ -1742,18 +1774,18 @@ def dafgsr(handle, recno, begin, end):
     :type begin: int
     :param end: Last word to read from record.
     :type end: int
-    :return: Contents of record.
-    :rtype: float
+    :return: Contents of request sub-record
+    :rtype: float numpy.ndarray
     """
     handle = ctypes.c_int(handle)
     recno = ctypes.c_int(recno)
     begin = ctypes.c_int(begin)
     end = ctypes.c_int(end)
-    data = ctypes.c_double()
+    data = stypes.emptyDoubleVector(125)
     found = ctypes.c_bool()
-    libspice.dafgsr_c(handle, recno, begin, end, ctypes.byref(data),
+    libspice.dafgsr_c(handle, recno, begin, end, data,
                       ctypes.byref(found))
-    return data.value, found.value
+    return stypes.cVectorToPython(data)[:end.value+1-begin.value], found.value
```

Comment:
See https://naif.jpl.nasa.gov/pub/naif/toolkit_docs/C/cspice/dafgsr_c.html, specifically https://naif.jpl.nasa.gov/pub/naif/toolkit_docs/C/cspice/dafgsr_c.html#Exceptions

This routine retrieves a vector of double precision values from [beg:end], where beg and end are limited to 1 and 128, respectively.  So if you ask for beg=1 and end=10, or beg=11 and end=20, you only get 10 values, and it puts item beg (one-based) into data[0], and item end (again, one-based) into data[1+end-beg].  So the `data = stypes.emptyDoubleVector(125)` could be `data = stypes.emptyDoubleVector(1+end-beg)`

Now that I look at it again, the data vector should be sized to 128, not to 125; I was assuming the caller would be getting only summary record data, which exclude the first three bytes.  However, there is nothing stopping the caller from retrieving all 128 values.

The dafgsr_c internal C code protects the user from bad beg:end pairs e.g. overflow as in beg=1, end=1280.  Hmm, maybe there should also be some intelligence overflow about in the Python wrapper.

There is also the issue that the beg:end interval is specified in 1-based form; changing that to be begminus1:end in the Python code resulting in begminus1+1:end here would be Pythonic.  However, since NAIF chose not to do that when they wrote the C wrapper, it makes sense to continue that choice for the Python wrapper.  Otherwise, the NAIF CSPICE documentation, which is all the SpiceyPy user has, would become misleading for that SpiceyPy user.